modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/MediaHandler.java

```diff
@@ -175,19 +175,26 @@ private void verifyMediaScope( final String projectContext, final RepositoryId r
         {
             if ( MEDIA_SCOPE_DELIMITER_PATTERN.splitAsStream( mediaServiceScope ).map( String::trim ).noneMatch( projectContext::equals ) )
             {
-                throw WebException.notFound( "Not a valid media url pattern" );
+                throw createNotFoundException();
             }
         }
-        else
+
+        if ( webRequest.getRawPath().startsWith( "/api/" ) )
         {
-            if ( webRequest instanceof final PortalRequest portalRequest )
-            {
-                if ( portalRequest.isSiteBase() &&
-                    !( repositoryId.equals( portalRequest.getRepositoryId() ) && branch.equals( portalRequest.getBranch() ) ) )
-                {
-                    throw WebException.notFound( "Not a valid media url pattern" );
-                }
-            }
+            return;
+        }
+
+        if ( !( webRequest.getRawPath().startsWith( "/site/" ) || webRequest.getRawPath().startsWith( "/admin/site/" ) ) )
```

Comment:
This code doesn't belong here . It should have been caught earlier that url doesn't match a supported one

Reply:
Actually that is done here https://github.com/enonic/xp/pull/10870/files/fd2dab8d8e5c17d8d68a4d39d4d56958d36f0f32#diff-54d9f82dd14e1847a6cfeca0bc4fc2fa72bc35df16da8d9eaeaf2a944f535dacR148